packages/fxa-settings/src/components/VerifiedSessionGuard/index.tsx

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { gql, useQuery } from '@apollo/client';
+
+const GET_SESSION = gql`
+  query GetSession {
+    session {
+      verified
+    }
+  }
+`;
+
+export const VerifiedSessionGuard = ({
+  guard,
+  children,
+}: {
+  guard: React.ReactNode;
+  children: React.ReactNode;
+}) => {
+  const { error, data } = useQuery(GET_SESSION, { fetchPolicy: 'cache-only' });
+
+  if (error) {
+    // not sure if this is necessary
+    throw error;
+  }
+
+  if (!data.session.verified) {
+    return <>{guard}</>;
+  } else {
+    return <>{children}</>;
+  }
```

Comment:
Could simplify this a bit with

`return data.session.verified ? children : guard`

But I'm Eh-Okay with this.

Reply:
that doesn't work with `ReactNode` but does with `ReactElement` which ought to be ok for our use